src/rust/src/wallet.rs

```diff
@@ -41,7 +43,15 @@ pub struct DecryptedNote {
     memo: [u8; 512],
 }
 
-struct TxNotes {
+/// A data structure tracking the note data that was decrypted from a single transaction,
+/// along with the height at which that transaction was discovered in the chain.
+#[derive(Debug, Clone)]
+pub struct TxNotes {
+    /// The block height containing the transaction from which these
+    /// notes were decrypted.
+    tx_height: BlockHeight,
```

Comment:
In looking into persistence, it looks like adding the block height information here is going to be problematic, because we don't have the block height available when we load the WalletTx data from disk. Essentially, the wallet restore process doesn't go through `AddToWalletIfInvolvingMe`.

Reply:
The solution that I'm going to pursue here: 
* tx_height becomes `Option<BlockHeight>`
* remove `block_height` from `add_notes_from_bundle` arguments
* update this data in `append_bundle_commitments`
* for initial wallet load, add a `set_tx_mined_height(txid: TxId, height: BlockHeight)` method that can be called from `InitiLoadWallet` on the C++ side.
* on InitLoadWallet, after `pindexRescan` has been determined, iterate over the wallet transactions and call `set_tx_mined_height` for each one.